From 4dcd65913c4ec56bc6fe906a6c80c209666b3828 Mon Sep 17 00:00:00 2001
From: Matthias Yzusqui <myzb.dev@gmail.com>
Date: Mon, 31 Oct 2016 17:06:09 +0100
Subject: [PATCH] sepolicy: allow sdcard access via sdcardfs

Change-Id: I755dc4d8adc9b8888c20bf8841e18ed4cf4ed69e
---

diff --git a/app.te b/app.te
index 4462062..1c675a4 100644
--- a/app.te
+++ b/app.te
@@ -184,11 +184,17 @@
 allow appdomain mnt_user_file:dir r_dir_perms;
 allow appdomain mnt_user_file:lnk_file r_file_perms;
 
-# Read/write visible storage
+# Read/write visible storage (vfat/f2fs/ext4/exfat) via fuse/sdcardfs
 allow appdomain fuse:dir create_dir_perms;
 allow appdomain fuse:file create_file_perms;
 allow appdomain sdcardfs:dir create_dir_perms;
 allow appdomain sdcardfs:file create_file_perms;
+allow appdomain vfat:dir create_dir_perms;
+allow appdomain vfat:file create_file_perms;
+allow appdomain sdcard_posix:dir create_dir_perms;
+allow appdomain sdcard_posix:file create_file_perms;
+allow appdomain fuseblk:dir create_dir_perms;
+allow appdomain fuseblk:file create_file_perms;
 
 # Access OBBs (vfat images) mounted by vold (b/17633509)
 # File write access allowed for FDs returned through Storage Access Framework
diff --git a/untrusted_app.te b/untrusted_app.te
index 6bc6843..b1c90d6 100644
--- a/untrusted_app.te
+++ b/untrusted_app.te
@@ -181,7 +181,10 @@
 neverallow untrusted_app {
   fs_type
   -fuse                     # sdcard
+  -fuseblk                  # sdcard
   -sdcardfs                 # sdcard
+  -sdcard_posix             # sdcard
+  -vfat                     # sdcard
   file_type
   -app_data_file            # The apps sandbox itself
   -media_rw_data_file       # Internal storage. Known that apps can
